src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  const currentState = { ...state };
+
+  actions.forEach((actionObj) => {
+    switch (actionObj.type) {
+      case 'addProperties':
+        Object.assign(currentState, { ...actionObj.extraData });
+        break;
+      case 'removeProperties':
+        actionObj.keysToRemove.forEach((key) => {
+          delete currentState[key];
+        });
+        break;
```

Comment:
you're creating 2 loops - both in `removeProperties` and `clear`, I think it would be worth to create a separate function - just for deleting object fields, and then just pass the keys you want to delete